File: examples/aoc/2025/1/main.py
def mod(x: int, y: int) -> int:
    return (x % y + y) % y


def part_1(numbers: list[int]) -> None:
    dial = 50
    count = 0
    for num in numbers:
        dial = mod(dial + num, 100)
        if dial == 0:
            count += 1
    print(f"part_1: {count}")


def part_2(numbers: list[int]) -> None:
    dial = 50
    count = 0
    for num in numbers:
        if dial > 0 and num < 0 and dial + num < 0:
            count += 1
        count += max(abs(dial + num) - 1, 0) // 100
        dial = mod(dial + num, 100)
        if dial == 0:
            count += 1
    print(f"part_2: {count}")


def parse(line: str) -> int:
    if not line:
        return 0
    val = int(line[1:])
    return val * -1 if line[0] == "L" else val


def main():
    with open("input.txt") as f:
        numbers = [parse(line) for line in f.read().split("\n")]
    part_1(numbers)
    part_2(numbers)


if __name__ == "__main__":
    main()

File: examples/aoc/2025/1/main.lua
local function mod(x, y)
    return (x % y + y) % y
end

local function part_1(numbers)
    local dial = 50
    local count = 0
    for _, num in ipairs(numbers) do
        dial = mod(dial + num, 100)
        if dial == 0 then count = count + 1 end
    end
    print("part_1: " .. count)
end

local function part_2(numbers)
    local dial = 50
    local count = 0
    for _, num in ipairs(numbers) do
        if dial > 0 and num < 0 and dial + num < 0 then
            count = count + 1
        end
        local adj = math.abs(dial + num) - 1
        count = count + (adj >= 0 and adj // 100 or 0)
        dial = mod(dial + num, 100)
        if dial == 0 then count = count + 1 end
    end
    print("part_2: " .. count)
end

local function parse(line)
    if line == "" then return 0 end
    local val = tonumber(line:sub(2))
    if line:sub(1, 1) == "L" then return -val else return val end
end

local function main()
    local f = io.open("input.txt", "r")
    local content = f:read("*a")
    f:close()
    local numbers = {}
    for line in content:gmatch("[^\n]*") do
        table.insert(numbers, parse(line))
    end
    part_1(numbers)
    part_2(numbers)
end

main()
